oak_remote_attestation_verification/src/verifier.rs

```diff
@@ -70,25 +79,161 @@ impl AttestationVerifier for InsecureAttestationVerifier {
+        let default_layer_verifier = opts
+            .default_layer_verification_option
+            .as_ref()
+            .map(|opt| Box::new(AttestationLayerVerifier::create(opt)));
+        let mut layer_verifiers = BTreeMap::new();
+        opts.layer_verification_options
+            .iter()
+            .for_each(|(name, opt)| {
+                layer_verifiers.insert(
+                    name.to_owned(),
+                    Box::new(AttestationLayerVerifier::create(opt)),
+                );
+            });
+        // TODO(#3641): Set other verifiers, e.g., for verifying TEE certificate.
+        let verifiers = Vec::new();
+        Self {
+            verifiers,
+            layer_verifiers,
+            default_layer_verifier,
+        }
+    }
+}
+
+impl AttestationVerifier for ConfigurableAttestationVerifier {
+    fn verify(
+        &self,
+        evidence: &AttestationEvidence,
+        endorsement: &AttestationEndorsement,
+        reference_value: &ReferenceValue,
+    ) -> anyhow::Result<()> {
+        for verifier in &self.verifiers {
+            verifier.verify(evidence, endorsement, reference_value)?;
+        }
+        // TODO(#3641): Update accordingly once #4074 is completed.
+        for verifier in self.layer_verifiers.values() {
+            verifier.verify(evidence, endorsement, reference_value)?;
+        }
+        //  TODO(#3641): Run the following only for layers not in `layer_verifiers`.
+        if let Some(verifier) = &self.default_layer_verifier {
+            verifier.verify(evidence, endorsement, reference_value)?;
+        }
+        Ok(())
+    }
+}
+
+/// Verification logic for verifying evidence from a single DICE layer. This verifier can be
+/// configured using an instance of LayerVerificationOptions.
+pub struct AttestationLayerVerifier {
+    verifiers: Vec<Box<dyn AttestationVerifier>>,
+    binary_verifiers: BTreeMap<String, Vec<Box<dyn AttestationVerifier>>>,
+    default_binary_verifier: Option<Vec<Box<dyn AttestationVerifier>>>,
+}
+
+impl AttestationLayerVerifier {
+    pub fn create(opts: &LayerVerificationOptions) -> AttestationLayerVerifier {
+        let default_binary_verifier = opts
+            .default_transparency_verification_options
+            .as_ref()
+            .map(create_transparency_verifier);
+
+        let mut binary_verifiers = BTreeMap::new();
+        opts.transparency_verification_options
+            .iter()
+            .for_each(|(name, opt)| {
+                binary_verifiers.insert(name.to_owned(), create_transparency_verifier(opt));
+            });
+        // TODO(#3641): Set other verifiers, e.g., reference value verifiers.
+        let verifiers = Vec::new();
+        Self {
+            verifiers,
+            binary_verifiers,
+            default_binary_verifier,
+        }
+    }
+}
+
+impl AttestationVerifier for AttestationLayerVerifier {
+    fn verify(
```

Comment:
I don't think AttestationVerifier is a sutable trait to implement here. The layer verifier must verify a layer in the evidence, but this signature takes the entire evidence. Also, a single layer might have multiple endorsements and multiple reference values.